radiation/TwoStreamMLPEMod.F90

```diff
@@ -0,0 +1,1685 @@
+Module TwoStreamMLPEMod
+
+  ! This module holds the routines to calculate two-tream
+  ! radiation scattering of vegetation in
+  ! "M"ultiple "L"ayers with "P"arellel "E"lements "MLPE"
+  !
+  ! In summary,
+  ! there may numerous canopy layers. In each canopy layer,
+  ! plant media for different functional types are grouped
+  ! so that they inhabit their own exclusive footprint
+  ! within the layer. Within these exclusive functional
+  ! columns, there are further sub-layer discretizations,
+  ! which are organized by top-down integrated vegetation
+  ! area index.
+  !
+  ! Note that there is a separate allocation and call
+  ! sequence for each broad band.  In other words, the
+  ! two_stream_type is instantiated for each broad band.
+  !
+  ! 
+  !
+  ! Assumptions: band index 1 = visible (vis)
+  !                         2 = near infrared (nir)
+  !                         3 = thermal (not used at the moment)
+  !
+
+  use shr_log_mod   , only: errMsg => shr_log_errMsg
+  use shr_sys_mod   , only: shr_sys_abort
+  
+  implicit none
+  private
+
+  integer, parameter :: r8 = selected_real_kind(12)
```

Comment:
Potentially, I wanted this module to be completely independent of FATES, in case it was adopted by the HLMs as well.  But, seeing how the HLMs would also have access to FatesConstantsMod, and that has no dependencies, it's probably fine to do this. I'll make the change.